File: packages/ash-php/src/Core/ContextPublicInfo.php
<?php

declare(strict_types=1);

namespace Ash\Core;

/**
 * Public context info returned to client.
 */
final class ContextPublicInfo
{
    /**
     * @param string $contextId Opaque context ID
     * @param int $expiresAt Expiration timestamp (ms epoch)
     * @param AshMode $mode Security mode
     * @param string|null $nonce Optional nonce (if server-assisted mode)
     */
    public function __construct(
        public readonly string $contextId,
        public readonly int $expiresAt,
        public readonly AshMode $mode,
        public readonly ?string $nonce = null,
    ) {
    }
}

File: packages/ash-dotnet/src/Ash.Core/ContextPublicInfo.cs
// ASH was developed by 3maem Co. | 12/31/2025

namespace Ash.Core;

/// <summary>
/// Public context info returned to client.
/// </summary>
public sealed class ContextPublicInfo
{
    /// <summary>
    /// Opaque context ID.
    /// </summary>
    public required string ContextId { get; init; }

    /// <summary>
    /// Expiration timestamp (ms epoch).
    /// </summary>
    public required long ExpiresAt { get; init; }

    /// <summary>
    /// Security mode.
    /// </summary>
    public required AshMode Mode { get; init; }

    /// <summary>
    /// Optional nonce (if server-assisted mode).
    /// </summary>
    public string? Nonce { get; init; }
}
